### PR TITLE
init Feature/AP5 92

### PR DESCRIPTION
**Pull request by jasonephraim**
There is some room for refactoring on implementationTimeline.js to better-generate the week number elements but I'm concerned about time before demo

### CODE REVIEW

File: rigex-front-end/src/components/rigScheduler/implementationTimelineStyle.scss [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    .table-heading-container {
+      font-weight: 900;
+      font-size: 16px;
+
+      .table-title-row {
+        .table-title {
+          font-size: 20px;
+          padding: 10px 30px;
+          border: 0.5px solid #c5c5c5;
```
Comment (jeremy-roussel) [inline rigex-front-end/src/components/rigScheduler/implementationTimelineStyle.scss]: colors belong in colors.scss file

**Review by jeremy-roussel — APPROVED** [review]

File: rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+  const DeliveryPhaseHeading = () => {
+    const totalDurationInHours = deliveryPhases.data.deliveryPhase.reduce(
+      function (previous, current) {
+        return parseFloat(previous + current.durationInHours);
+      },
+      0
+    );
+    const totalDurationInWeeks = totalDurationInHours / 40;
+    const ab = [1, totalDurationInWeeks];
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js]: what is ab? Is there a better name for this?

File: rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        <td className="pre-delivery-content">
+          {deliveryPhases.data.preDeliveryPhase[0].text}
+        </td>
+      </tr>
+    );
+  };
+
+  const DeliveryPhaseHeading = () => {
+    const totalDurationInHours = deliveryPhases.data.deliveryPhase.reduce(
+      function (previous, current) {
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js]: can you use an arrow function

File: rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    );
+    const totalDurationInWeeks = totalDurationInHours / 40;
+    const ab = [1, totalDurationInWeeks];
+    const start = ab[0];
+    const end = ab[1];
+    const interval = (end - start) / 5;
+    const out = Array(5)
+      .fill()
+      .map((_, i) => [start + i * interval, start + (i + 1) * interval]);
+    let sortedOut = out.map(function (subArray) {
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js]: wherever you have the function word let's use arrow functions instead

File: rigex-front-end/src/components/rigScheduler/implementationTimelineStyle.scss [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        border-bottom: 0.5px solid #c5c5c5;
+        border-radius: 0 0 12px 0;
+      }
+
+      tr:nth-child(6n + 1) td .bar {
+        background-color: #ffb200;
+        border-color: #ffb200;
+      }
+      tr:nth-child(6n + 2) td .bar {
+        background-color: #ffe800;
```
Comment (BarretteBanner) [inline rigex-front-end/src/components/rigScheduler/implementationTimelineStyle.scss]: your colors should come from a colors.scss file or something of that nature

File: rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    const durations = [1, totalDurationInWeeks];
+    const start = durations[0];
+    const end = durations[1];
+    const interval = (end - start) / 5;
+    const out = Array(5)
+      .fill()
+      .map((_, i) => [start + i * interval, start + (i + 1) * interval]);
+    let sortedOut = out.map((subArray) =>
+      subArray.map((elem) => Number(elem.toFixed(0)))
+    );
```
Comment (mcraepv) [inline rigex-front-end/src/components/rigScheduler/ImplementationTimeline.js:34]: nice algo

**Review by mcraepv — APPROVED** [review]

**Pull request merged**